File: nodejs/src/conDependencyInjection/conLibreria/UserRepository.js
import {User} from "./User";

export class UserRepository {
    users = []

    constructor(opts) {
        this.logger = opts.logger
    }

    create(nombre, apellido) {
        const user = new User(Math.random(), nombre, apellido)
        this.users.push(user);
        this.logger.log(`user con nombre ${nombre} y apellido ${apellido} creado`)
        return user
    }

    findById(id) {
        return this.users.find(user => user.id === id)
    }
}

File: kotlin/src/main/kotlin/com/di/kotlin/di/solo/UserRepository.kt
package com.di.kotlin.di.solo

import com.di.kotlin.ILogger
import com.di.kotlin.IUserRepository
import com.di.kotlin.model.User

class UserRepository(private val logger: ILogger) : IUserRepository {

    private val users: MutableList<User> = mutableListOf();

    override fun findById(id: Long) : User? {
        return users.find { it.id == id }
    }

    override fun createUser(nombre: String, apellido: String) : User{
        this.logger.log("user con nombre $nombre y apellido $apellido creado")
        val user = User(Math.random().toLong(), nombre, apellido)
        users.add(user)
        return user
    }

}